QoL Features for the List Item Modal

GODrums opened the pull request:
## Motivation

The new listing modal only contains a small subset of features from the website. Users may want more customization options.

## Description

This PR adds new elements to the "list item"-modal to provide higher feature parity with the CSFloat website. This includes:
- a link to get to CSFloat's /sell-page in case someone doesn't want to use / doesn't trust the modal
- an informative banner linking CSFloat's blog article for auctions (similar to `/sell`)
- an assessment label for the price (similar to `/sell`)
- an option to create a private listing
- a "search on market"-link to refer back to the market listings website
- option to add a description to the listing (hidden behind a plus per default)

## Screenshot
![image](https://github.com/user-attachments/assets/a377d3a4-64ae-4e5f-a86a-5d5c51e5f4d9)
After clicking the "plus" for the description:
![image](https://github.com/user-attachments/assets/87a7ea9e-05a2-45f6-b648-e50e3ccf66ab)

Step7750 commented on src/lib/components/inventory/list_item_modal.ts:
```diff
@@ -8,11 +8,14 @@ import {ClientSend} from '../../bridge/client';
-import {FetchPendingTrades} from '../../bridge/handlers/fetch_pending_trades';
 import {FetchCSFloatMe} from '../../bridge/handlers/fetch_csfloat_me';
 
+// SVG icons for percentage assessment
```
Perhaps we can move these icons to a central `icons.ts` file either in this directory or in a more global place?

Step7750 commented on src/lib/components/inventory/list_item_modal.ts:
```diff
@@ -354,17 +382,20 @@ export class ListItemModal extends FloatElement {
                                 >
                                     View Your Stall on CSFloat
                                 </a>
+                                <a href="${this.searchUrl}" target="_blank" class="base-button secondary-button">
```
They may want a way to search these items without having to list the item first -- perhaps there's a way to put a UI element somewhere (maybe to the right of the recommended price above the input).

Step7750 commented on src/lib/components/inventory/list_item_modal.ts:
```diff
@@ -468,15 +544,44 @@ export class ListItemModal extends FloatElement {
-                                      ${this.recommendedPrice && this.customPrice
-                                          ? Math.round((this.customPrice / this.recommendedPrice) * 100)
-                                          : 100}%
+                                      <label>Visibility</label>
```
Toggling visibility to private is pretty rare -- this UI component takes a large amount of space given how infrequent it'll be used.

Perhaps we can right-align it on the same line as the "Visibility" text and make the buttons much smaller?

Step7750 reviewed (changes requested):
Thanks for these QOL features @GODrums !

GODrums commented:
Additionally, I added listing support for the new music kits and pins now in 74e0e03.
And it seemed like the charms-detection was broken as well. With 10ac9ad it should be fixed again.

GODrums commented:
Added a new description field now. Per default this it is hidden behind a "plus"-button to avoid overwhelming the user.

I also fixed another issue from the original modal PR, which caused the slider to not be updated when manually entering a price (35a60b4).

Step7750 reviewed (approved):
This is great @GODrums ! LGTM
